### DISCUSSION TITLE
tr-lang 0.4.0 - `Hakan` upgrades people, upgrades

### DISCUSSION THREAD

**Discussion by kaiserthe13th** [kaiserthe13th/tr-lang#32] [Announcements | Duyurular]
# English :gb: :us: 
- The language now uses infix notation instead of postfix(reverse polish) notation although you can still use postfix by wrapping operators in parentheses
- Better error reporting (In Progress)
- Added a Windows Executable
- Added a MacOS Executable (M1)
- Added RPM File
- new module system
- block, and none datatypes (list also exists as a datatype but isn't implemented)
- new accessor operator
- interactive mod
- command mode
- new error api (yes, again)
- tr-lang as a library
- simple test runner
- newer and better install script
- editor support
- a bunch of bugfixes

> I'll be taking a break while the project gets evaluated at TÜBİTAK.
> The evaluation will end in march so not for that long really. Bye!

# Türkçe :tr: 

- Dil artık postfix(ters leh) notasyonu yerine infix notasyon kullanıyor, ancak hala postfix kullanmak istiyorsanız operatörlerinizi parentez içine almanız yeter
- Daha iyi hata raporlama (Yapım aşamasında)
- Bir Windows Executable eklendi
- Bir MacOS Executable eklendi
- RPM Dosyası eklendi
- yeni modül sistemi
- blok, ve hiç veri tipleri (liste'de var ama uygulanmış değil)
- yeni erişim operatörü
- interaktif mod
- komut modu
- yeni hata api'si (evet, yine)
- tr-lang'ın kütüphane olarak kullanılabilir olması
- basit test yürütücü
- yeni ve daha iyi indirme programı
- editor desteği
- birkaç tane bug düzeltilmesi

> Proje TÜBİTAK tarafından değerlendirilirken mola vericem.
> Değerlendirme mart'ta bitecek o yüzden pek uzun sürmeyecek. Görüşürüz!

<hr /><em>This discussion was created from the release <a href='https://github.com/kaiserthe13th/tr-lang/releases/tag/tr-lang-0.4.0'>tr-lang 0.4.0 - `Hakan` upgrades people, upgrades</a>.</em>

**Comment by kaiserthe13th** [comment]
## Turkish

Merhaba, tr-lang'ın geleceği hakkında konuşmak için buradayım.

Şu anda farklı bir dil olan çok daha fazla emek gerektiren özelliklere sahip olacak olan "Krepé" üzerine çalışmaktayım. Ve bu yüzden sonraki güncelleme belirsiz bir süre boyunca duraksamıştır. Bu yeni dilin gelecekte tr-lang'la ortak bir ara temsil dili paylaşması planlanan bir derleyici olacak. Bununla beraber, Krepé'yi kullanarak tr-lang'ı sıfırdan yeniden yazmayı planlıyorum.

Yeni tr-lang'ın daha satır satır işleyen, bazo işleçlere özel kurallar dışında zircirlenmesiz bir sözdizime sahip olması planlanıyor. Yorumlanabilecek (ortak ara temsil dili üzerinden) ya da önden derlenebilecek olması isteniyor.

## English

Ok, let's talk about the future of tr-lang.

Right now I am working on a different language with much more intricate features and syntax "Krepé". And because of that the next update was put on hold indefinitely. The new language is planned to be a compiler that's hopefully going to have a common intermediate language with tr-lang in the future. With that said, I plan to rewrite tr-lang from the ground up in Krepé.

The new tr-lang will have a more line by line non-concatenative approach to syntax with a few quirks for operators. It's wanted that it be able to be interpreted (from the common IR) or AOT compiled.

**Reply by volkantash** [reply]
👀
